tests/Main.hs

```diff
@@ -0,0 +1,67 @@
+import Data.Proxy       (Proxy (..))
+import System.Directory (doesFileExist)
+import System.FilePath  (takeDirectory, FilePath, (</>))
+import Test.Tasty
+import Test.Tasty.Golden  (findByExtension)
+import Test.Tasty.Options (IsOption (..), OptionDescription (Option))
+import Test.Tasty.Program (testProgram)
+
+
+main :: IO ()
+main = do
+  input <- testInput
+  defaultMainWithIngredients ings . askOption $ \args -> mkTestGrp args input
+    where
+      ings = includingOptions [Option (Proxy :: Proxy Args)] :
+        defaultIngredients
+
+newtype Args = A String
+
+instance IsOption Args where
+  defaultValue = A ""
+  parseValue = Just . A
+  optionName = return "sixten-args"
+  optionHelp = return "Arguments to supply to the sixten binary."
+
+testRootDir :: FilePath
+testRootDir = "tests"
+testName :: FilePath -> FilePath
+testName = drop (length testRootDir + 1)
+
+groupByKey :: Ord b => (a -> b) -> [a] -> [(b, [a])]
```

Comment:
I think this could be `Eq`. Nice, useful function!

Reply:
Good catch! Initially I was going to sort stuff but I forgot to fix the signature.